columna GDE

MariaNavarro90 opened the pull request:
# Información de la Tarea
Vincular el #485 

# Resumen de la Solución

- Se agregó una nueva columna "Número GDE" a la tabla de documentos de admisiones que permite a los técnicos de comedor y superadministradores asignar números de expediente GDE después de cargar documentos en sistemas externos
- La funcionalidad incluye validaciones de permisos, auditoría completa y una interfaz de usuario intuitiva con edición inline
- Solo documentos en estado "Aceptado" pueden tener números GDE asignados.

## Información Adicional

- Los números GDE son críticos para el seguimiento de expedientes en sistemas externos del gobierno
- La implementación sigue el patrón establecido en el proyecto con servicios, vistas AJAX y validaciones de seguridad
- Se mantiene compatibilidad con versiones anteriores y no afecta funcionalidades existentes
- Incluye logging completo para auditoría y troubleshooting

# Descripción de los Cambios

## Cambios

- **Modelo**: Agregado campo `numero_gde` en `ArchivoAdmision` con validaciones y help text
- **Servicio**: Nuevo método `actualizar_numero_gde_ajax()` con validaciones de permisos y estado
- **Vista**: Nueva vista `actualizar_numero_gde_archivo()` para manejo de peticiones AJAX
- **Template**: Interfaz de edición inline con modos vista/edición y validaciones de permisos
- **JavaScript**: Funciones para manejo de edición inline, validaciones y feedback visual
- **URLs**: Nueva ruta para la funcionalidad AJAX de actualización
- **CSS**: Estilos para la interfaz de edición con hover effects y responsive design

# Cómo Testear los Cambios

## Pruebas Manuales

### Como Técnico de Comedor

1. Acceder a una admisión con documentos en estado "Aceptado"
2. Hacer clic en "Sin número GDE" o en el ícono de edición
3. Ingresar un número GDE y guardar
4. Verificar que se muestre el número en verde y sea editable nuevamente

### Otras Validaciones

- **Como usuario sin permisos**: Verificar que la columna GDE sea de solo lectura
- **Validación de estados**: Confirmar que solo documentos "Aceptados" muestren la opción de edición
- **Feedback visual**: Comprobar toasts de éxito, colores apropiados y transiciones suaves
- **Manejo de errores**: Intentar editar sin permisos y verificar mensajes de error apropiados

# Capturas de Pantalla
- 
<img width="2034" height="582" alt="image" src="https://github.com/user-attachments/assets/999768a2-b3d5-46fb-80f9-1684c4f63e35" />

- 
<img width="361" height="68" alt="image" src="https://github.com/user-attachments/assets/c17acd28-a6cb-4ed7-b90e-4c96c9b14d8b" />

- <img width="1971" height="373" alt="image" src="https://github.com/user-attachments/assets/6e267a6b-cea9-47e7-b44c-f01e66431199" />

MariaNavarro90 committed:
feature: agregado 'numero_gde' en el modelo ArchivoAdmision

MariaNavarro90 committed:
feature: agregada lógica para actualizar número GDE a través de AJAX

MariaNavarro90 committed:
feature: agregada ruta AJAX para actualizar número GDE

MariaNavarro90 committed:
se agregó un endpoint AJAX para actualizar 'numero_gde' con manejo de respuesta

MariaNavarro90 committed:
se agregó manejo de errores y retroalimentación visual para la función actualizarNumeroGDE

MariaNavarro90 committed:
se agregó input del número GDE y la retroalimentación de éxito en el formulario del técnico de admisiones

MariaNavarro90 committed:
feature: agregado campo 'numero_gde' en el contexto de documentos en AdmisionService

MariaNavarro90 committed:
implementada la funcionalidad de edición para el número GDE con retroalimentación visual

MariaNavarro90 committed:
se implementó la funcionalidad de edición para el número GDE con una interfaz de usuario mejorada y retroalimentación visual

MariaNavarro90 committed:
se mejoró la verificación de permisos para la actualización del número GDE,

MariaNavarro90 committed:
mejorada la verificación de permisos para la edición del número GDE y optimizada la lógica de acceso a documentos, eliminados logs ya no necesarios.

MariaNavarro90 committed:
migracion

MariaNavarro90 committed:
black

MariaNavarro90 committed:
djlint

MariaNavarro90 committed:
Merge branch 'development' of https://github.com/dsocial118/SISOC-Backoffice into 485_GDE

dsocial118 commented on admisiones/services/admisiones_service.py:
```diff
@@ -480,3 +488,76 @@ def get_admision_instance(admision_id):
+            archivo = get_object_or_404(ArchivoAdmision, id=documento_id)
+
+            # Verificar que el documento esté en estado "Aceptado"
+            if archivo.estado != "Aceptado":
```
Bloqueante: Curiosamente no existe el estado "Aceptado" en los choices, hay que añadirlo

dsocial118 commented on static/custom/js/admisionesactualizarestado.js:
```diff
@@ -89,3 +89,130 @@ function actualizarEstado(selectElement) {
+        input.value = numeroGDE || '';
+        
+        // Actualizar el contenido de la vista
+        if (numeroGDE) {
```
Bloqueante: El HTML en texto nos deja expuestos a ataques XSS. Lo mejor seria escaparlo con Js con un CreateElement

dsocial118 commented on admisiones/services/admisiones_service.py:
```diff
@@ -480,3 +488,76 @@ def get_admision_instance(admision_id):
             return None
+
+    @staticmethod
+    def actualizar_numero_gde_ajax(request):
```
Sugerencia: Podemos poner un docstring aca que esta media compleja la funcion 👉👈

dsocial118 commented on admisiones/templates/admisiones_tecnicos_form.html:
```diff
@@ -391,6 +392,54 @@ <h2 class="card-title fw-bold fs-3">Documentos requeridos</h2>
                                                                         {% endif %}
                                                                     </td>
+                                                                    <td id="gde-{{ doc.id }}">
+                                                                        {% if doc.estado == "Aceptado" and request.user.is_superuser or doc.estado == "Aceptado" and request.user|has_group:"Tecnico Comedor" %}
```
Sugerencia: Esto se puede simplificar con:
`{% if doc.estado == "Aceptado" and (request.user.is_superuser or request.user|has_group:"Tecnico Comedor") %}`

MariaNavarro90 committed:
Merge branch 'development' of https://github.com/dsocial118/SISOC-Backoffice into 485_GDE

MariaNavarro90 committed:
comentario PR: Bloqueante: Curiosamente no existe el estado "Aceptado" en los choices, hay que añadirlo

MariaNavarro90 committed:
comentario PR: Bloqueante: El HTML en texto nos deja expuestos a ataques XSS. Lo mejor seria escaparlo con Js con un CreateElement

MariaNavarro90 committed:
comentario PR: Sugerencia: Podemos poner un docstring aca que esta media compleja la funcion 👉👈

MariaNavarro90 committed:
comentario PR: Sugerencia: Esto se puede simplificar con:
{% if doc.estado == "Aceptado" and (request.user.is_superuser or request.user|has_group:"Tecnico Comedor") %}

MariaNavarro90 committed:
migracion

MariaNavarro90 committed:
black

MariaNavarro90 committed:
djlint

dsocial118 committed:
Merge branch 'development' into 485_GDE
